assets/scripts/fe/src/app/components/payroll/payroll-list/payroll-list.component.ts

```diff
@@ -29,4 +33,68 @@ export class PayrollListComponent implements OnInit {
     }
   }
 
-}
+  onToggle(item){
+    // Function trigger upon checkbox toggle
+
+    // Checks if item is already exist in mapCheckedPayroll 
+    if(this.payrollservice.mapCheckedPayroll.has(item.id)){
+      // if item is exist. Remove it from mapped data
+      this.payrollservice.mapCheckedPayroll.delete(item.id);
+    }
+    else{
+      // else, add the item to mapped data
+      this.payrollservice.mapCheckedPayroll.set(item.id, item)
+    } 
+  }
+
+  downloadPDF(){
+    // loops over each item of the mapped data
+    this.payrollservice.mapCheckedPayroll.forEach((value: boolean, key: string) => {
+      // get file name
+      const file_name = this.getFileName(value);
+      // call the download functionality with parameters of item id and file name
+      this.payrollservice.downloadPDF(key, file_name);
+    });
+  }
+
+  getFileName(payrollValue){
+    // Constructing the file name for the pdf
+    const report_phrase = `${payrollValue.date_from} to ${payrollValue.date_to}`;
+    const date_phrase = `${payrollValue.date_from} to ${payrollValue.date_to}`;
+    const employee_name = `${payrollValue.user.first_name} to ${payrollValue.user.last_name}`;
+    const file_name = `payroll of ${employee_name} ${report_phrase}`;
+    return `payroll of ${employee_name} ${date_phrase}`;;
+  }
+
+
+  sendPDF(){
+    let keys = Array.from( this.payrollservice.mapCheckedPayroll.keys() );
+
+    if (keys.length){
+      // sendingEmail means that we have to disable the button
+      this.sendingEmail = true;
+      // Reset message if it's still sending
+      this.emailCallbackMessage = "";
+      this.payrollservice.sendPayrollReport(keys)
+      .then(
+        data => {
+          console.log(data);
+          this.sendPDFGeneralCallback();
+          this.emailCallbackMessage = "Email is sent sucessfully.";
```

Comment:
bad. do not put static user messages in these scripts. keep template items into template and view items to views.